src/decoder_fluidsynth.cpp

```diff
@@ -0,0 +1,213 @@
+/*
+ * This file is part of EasyRPG Player.
+ *
+ * EasyRPG Player is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * EasyRPG Player is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with EasyRPG Player. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "system.h"
+#include "decoder_fluidsynth.h"
+
+
+#ifdef HAVE_FLUIDSYNTH
+
+// Headers
+#include <cassert>
+#include "audio_decoder.h"
+#include "output.h"
+#include "decoder_fluidsynth.h"
+
+#include <fluidsynth.h>
+
+FluidSynthDecoder::FluidSynthDecoder() {
+	music_type = "midi";
+
+	settings = new_fluid_settings();
+	fluid_settings_setstr(settings, "player.timing-source", "sample");
+	fluid_settings_setint(settings, "synth.lock-memory", 0);
+
+	fluid_settings_setnum(settings, "synth.gain", 0.6);
+	fluid_settings_setint(settings, "synth.polyphony", 32);
+
+	synth = new_fluid_synth(settings);
+	if (fluid_synth_sfload(synth, "easyrpg.soundfont", 1) == FLUID_FAILED) {
+		error_message = "Could not load soundfont.";
+		return;
+	}
+
+	fluid_synth_set_interp_method(synth, -1, 7);
+
+	double sample_rate = 0;
+	fluid_settings_getnum(settings, "synth.sample-rate", &sample_rate);
+	assert(sample_rate != 0);
```

Comment:
If this is coming from a runtime configuration, assert is not valid here. We have to handle the error by exiting. assert is only for bugs in code.

Reply:
It seems that line is already gone